engine/src/main/java/io/zeebe/engine/processing/bpmn/container/SubProcessProcessor.java

```diff
@@ -56,21 +57,16 @@ public void onActivating(
   public void onActivated(
       final ExecutableFlowElementContainer element, final BpmnElementContext context) {
 
+    final ExecutableStartEvent startEvent;
     if (element.hasNoneStartEvent()) {
       // embedded sub-process is activated
-      final var noneStartEvent = element.getNoneStartEvent();
-      stateTransitionBehavior.activateChildInstance(context, noneStartEvent);
-
+      startEvent = element.getNoneStartEvent();
     } else {
       // event sub-process is activated
-      final var startEvent = element.getStartEvents().get(0);
-      final var childInstanceKey =
-          stateTransitionBehavior.activateChildInstance(context, startEvent);
-
-      // the event variables are stored as temporary variables in the scope of the subprocess
-      // - move them to the scope of the start event to apply the output variable mappings
-      stateBehavior.transferTemporaryVariables(context, childInstanceKey);
+      startEvent = element.getStartEvents().get(0);
     }
```

Comment:
Hey @npepinpe thanks for the review. Actually the distinction was before made regarding event sub process (an event sub process can't have a none start event) and this check we also have in the applier now. Or do you think I missed something else?

Reply:
I'm thinking maybe I missed something instead? :sweat_smile: 

So it looks to me like previously, if the start event was a none event, we'd just activate it. If it was something else (e.g. message, timer) we'd copy temporary variables over.

Now, we check if the start event is in a sub process, but not its type? That's the part that lost me (but admittedly my knowledge of the engine is still rusty)